### PR TITLE
[convert-environment-variables-and-secrets-guide-stack-2] Convert env vars and secrets guide from repository to Definitions

### PR DESCRIPTION
**Pull request by schrockn**
Summary & Motivation: This moves over the environment variables and secrets guide to use Definitions. Note that this links to the
Dagster Crash course which is currently not ported to the new APIs.

Test Plan: Read

### CODE REVIEW

**Comment by schrockn** [comment]
Depends on https://github.com/dagster-io/dagster/pull/10774

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx [hunk 1]
```diff
@@ -145,33 +145,30 @@ Let's review what's happening here:
...
+        )
+    },
+)
 ```
 
 Let's review what's happening here:
 
-- Using <PyObject object="with_resources" /> adds the `github_api` resource to the assets in the repository. In assets, we'll use the `github_api` resource key to reference the resource.
 - Using the <PyObject object="configured" /> method on the `github_api` resource, we can pass configuration info to the resource. In this example, we're telling Dagster to load the `access_token` from the `GITHUB_ACCESS_TOKEN` environment variable.
+- We're adding that configured resource to our `Definitions` object to that it is available for our assets.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:171]:
```suggestion
- We're adding that configured resource to our <PyObject object="Definitions" /> object so it's available for our assets.
```

Adding in the `PyObject` component here, so we can link to the API docs when they're ready

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx [hunk 2]
```diff
@@ -189,53 +186,45 @@ In this example, we'll demonstrate how to use different I/O manager configuratio
...
-- Lastly, we use <PyObject object="with_resources" /> to:
-  - Provide the resources to the assets in the repository
-  - Parameterize usage of the resources based on the value of `DAGSTER_DEPLOYMENT`
+- We've created a dictionary of resource definitions, `resources`, named after our `local` and `production` environments. In this example, we're using a [Pandas Snowflake I/O manager](/\_apidocs/libraries/dagster-snowflake-pandas).
+- For both `local` and `production`, we used the <PyObject object="configured" /> method on `snowflake_pandas_io_manager` to provide environment-specific run configuration. Note the differences in configuration between `local` and `production`, specifically where environment variables were used.
+- Following the resources dictionary, we define the `deployment_name` variable, which determines the current executing environment. This variable defaults to `local`, ensuring that `DAGSTER_DEPLOYMENT=PRODUCTION` must be set to use the `production` configuration.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:227]:
```suggestion
- Following the `resources` dictionary, we define the `deployment_name` variable, which determines the current executing environment. This variable defaults to `local`, ensuring that `DAGSTER_DEPLOYMENT=PRODUCTION` must be set to use the `production` configuration.
```

Suggesting code formatting to explicitly refer to the `resources` key

**Review by erinkcochran87 — APPROVED** [review]
lgtm!

**Pull request merged**